### PR TITLE
Return the right codes for material notify api

### PR DESCRIPTION
**Pull request by zabil**
204 - No content success response when the material is notified.
404 - When material is not found
content-type as text

### CODE REVIEW

**Comment by arvindsv** [comment]
The change looks good to me. Was this the one we talked about trying to return a 202? Should the content-type be JSON or plain text?

**Comment by zabil** [comment]
yes. have to make that change too.

--
Typed using fat thumbs




On Mon, Jun 1, 2015 at 6:29 AM -0700, "Aravind SV" <notifications@github.com> wrote:












The change looks good to me. Was this the one we talked about trying to return a 202? Should the content-type be JSON or plain text?

**Comment by arvindsv** [comment]
Ok. Won't merge, then. Unless you plan for it to be in an other PR.

**Comment by zabil** [comment]
Made the changes.

File: server/test/unit/com/thoughtworks/go/server/materials/MaterialUpdateServiceTest.java [hunk 1]
```diff
@@ -188,6 +205,11 @@ public void shouldReturnImplementerOfSvnPostCommitHookAndPerformMaterialUpdate_W
         spyService.notifyMaterialsForUpdate(username, params, result);
         verify(svnPostCommitHookImplementer).prune(anySet(), eq(params));
         verify(spyService).updateMaterial(svnMaterial);
+
+        HttpLocalizedOperationResult acceptedResult = new HttpLocalizedOperationResult();
+        acceptedResult.accepted(LocalizedMessage.string("MATERIAL_SCHEDULE_NOTIFICATION_ACCEPTED"));
+
+        assertThat(result, is(acceptedResult));
```
Comment (ketan) [inline server/test/unit/com/thoughtworks/go/server/materials/MaterialUpdateServiceTest.java]: Was there a reason to not be consistent with the rest of the testcase -

```
        assertThat(result.isAccepted(), is(false));
        assertThat(result.httpCode(), is(HttpStatus.SC_ACCEPTED));
        assertThat(result.hasMessage(), is(true));
```

Comment (zabil) [inline server/test/unit/com/thoughtworks/go/server/materials/MaterialUpdateServiceTest.java]: The equals was not implemented earlier. We have to do that for checking the localizedmessage. Will change it consistent at other places and synchronize this.

**Pull request merged**